chameleon/views.py

```diff
@@ -96,26 +104,30 @@ def redirect_to_horizon(request):
 '''
     Create projects and add accounts for PI's to Keystone as needed during SSO
 '''
-def create_ks_project_set_pi(active_tas_projects, ks_charge_codes, user):
+def create_ks_project_set_pi(ks_admin, active_tas_projects, ks_charge_codes, user):
     for p in active_tas_projects:
         if p.pi.username == user.username and p.chargeCode not in ks_charge_codes:
-            update_user_keystone_project_membership(p.pi.username, p, add_member=True)
+            update_user_keystone_project_membership(ks_admin, p.pi.username, p, add_member=True)
 
 def sync_user_project_status(ks_admin, user, chameleon_projects):
     ks_user = ks_admin.users.list(name=user.username)
     active_projects_in_tas = chameleon_projects.get('active_projects')
-    ks_projects = []
+
     if ks_user:
         ks_projects = ks_admin.projects.list(user=ks_user[0])
+    else:
+        ks_projects = []
```

Comment:
?

Reply:
Ah, this must have been part of my "working through the code". Sometimes I move things to where I would expect to find them--I tend to avoid re-declaring variables, but it is a personal preference and syntactically is equivalent.

Reply:
:D